x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/fields/ExtractedField.java

```diff
@@ -88,12 +99,86 @@ public static ExtractedField newField(String alias, String name, ExtractionMetho
             DocumentField keyValue = hit.field(name);
             if (keyValue != null) {
                 List<Object> values = keyValue.getValues();
-                return values.toArray(new Object[values.size()]);
+                return values.toArray(new Object[0]);
             }
             return new Object[0];
         }
     }
 
+    private static class GeoField extends ExtractedField {
+
+        private final ExtractedField internalExtractor;
+        private final WellKnownText wkt = new WellKnownText();
```

Comment:
Should we make this static to avoid constructing it for every field?

Reply:
@dimitris-athanasiou yep! Good catch.